### ISSUE TITLE
code-update: activation review issues.

### ISSUE THREAD

**Issue by williamspatrick** [openbmc/openbmc#1843]
Minor issues logged in 
https://gerrit.openbmc-project.xyz/#/c/4795/6
```
activation.cpp
Line 27:
Need a mechanism to unsubscribe when the object goes away or subscription need is done.
Line 80:
Can you place the contents of these if-conditions into private member functions? This function is way too long.
```

**Comment by rfrandse** [comment]
<https://gerrit.openbmc-project.xyz/5557> PNOR Activation: Minor code-review fixes
Resolves  #1843 code-update: activation review issues.

**Comment by mtritz** [comment]
# **STATUS**
7/26: Ready for review.
